bin/src/Main.purs

```diff
@@ -953,24 +959,26 @@ mkFetchEnv args@{ migrateConfig, offline } = do
     Left _err -> die $ "Failed to parse selected package name, was: " <> show args.selectedPackage
 
   env <- mkRegistryEnv offline
-  { rootPath } <- ask
-  workspace <-
-    runSpago (Record.union env { rootPath })
-      (Config.readWorkspace { maybeSelectedPackage, pureBuild: args.pure, migrateConfig })
+  cwd <- Paths.cwd
+  { workspace, rootPath } <-
+    runSpago env
+      (Config.discoverWorkspace { maybeSelectedPackage, pureBuild: args.pure, migrateConfig } cwd)
+
+  FS.mkdirp $ rootPath </> Paths.localCachePath
+  FS.mkdirp $ rootPath </> Paths.localCachePackagesPath
+  logDebug $ "Workspace root path: " <> Path.quote rootPath
+  logDebug $ "Local cache: " <> Paths.localCachePath
```

Comment:
This is only semi-related to the change. I found that, before this change, Spago would _always_ create a `./.spago` directory as soon as it's launched, even if it's only asked to `registry search`. I moved this from `mkRegistryEnv` to `mkFetchEnv`, which seems to be logically always "in a workspace".